specs/AddHostObjectToScriptWithOrigins.md

```diff
@@ -0,0 +1,275 @@
+}
+
+[uuid(cfd00c6a-8889-47de-84b2-3016d44dbaf4), object, pointer_default(unique)]
+interface ICoreWebView2FrameNameChangedEventHandler : IUnknown {
+  /// Provides the result for the iframe name changed event.
+  /// No event args exist and the `args` parameter is set to `null`.
+  HRESULT Invoke([in] ICoreWebView2Frame * sender, [in] IUnknown * args);
+}
+```
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    /// CoreWebView2Frame provides direct access to the iframes information and use.
+     runtimeclass CoreWebView2Frame
+    {
+        /// The name of the iframe from the iframe html tag declaring it.
+        /// Calling this method fails if it is called after the iframe is deleted.
+        String Name { get; };
+
+        /// Raised when the iframe changes its window.name property
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Frame, Object> NameChanged;
+        /// The FrameDeleted event is raised when the iframe corresponding to this CoreWebView2Frame
+        /// object is removed or the document containing that iframe is destroyed
+        event Windows.Foundation.TypedEventHandler<CoreWebView2Frame, Object> FrameDeleted;
+
+        /// Add the provided host object to script running in the iframe with the
+        /// specified name for the list of the specified origins. The host object
+        /// will be accessible for this iframe only if the iframe's origin during
+        /// access matches one of the origins which are passed. The provided origins
+        /// will be normalized before comparing to the origin of the document.
+        /// So the scheme name is made lower case, the host will be punycode decoded
+        /// as appropriate, default port values will be removed, and so on.
+        /// This means the origin's host may be punycode encoded or not and will match
+        /// regardless.
+        /// If the iframe is declared with no src attribute its origin is considered
+        /// the same as the origin of the parent document.
+        /// List of origins will be treated as following:
+        /// 1. empty list - call will fail and no host object will be added for the
+        /// iframe;
+        /// 2. list with origins - during access to host object from iframe the
+        /// origin will be checked that it belongs to this list;
+        /// 3. list with "*" element - host object will be available for iframe for
+        /// all origins. We suggest not to use this feature without understanding
+        /// security implications of giving access to host object from from iframes
+        /// with unknown origins.
+        /// Calling this method fails if it is called after the iframe is deleted.
+        /// \snippet ScenarioAddHostObject.cpp AddHostObjectToScriptWithOrigins
+        /// For more information about host objects navigate to [AddHostObjectToScript]
+        void AddHostObjectToScriptWithOrigins(String name, Object rawObject, String[] origins);
```

Comment:
`String[]` or `IVector<String>`?

Reply:
* IEnumerable for C#
* IIterable for IDL3
* remains pointer+size for COM